feat(baremetal): review docs (#3561)
scaleway/docs-content: bare-metal/elastic-metal/how-to/configure-flexible-ipv6.mdx

---
meta:
  title: How to configure a flexible IPv6 on an Elastic Metal server
  description: Learn how to configure a flexible IPv6 on a Scaleway Elastic Metal server.
content:
  h1: How to configure a flexible IPv6 on an Elastic Metal server
  paragraph: This guide explains how to configure a flexible IPv6 on a Scaleway Elastic Metal server.
tags: configuration, flexible-ipv6, ipv6, elastic-metal
dates:
  validation: 2024-08-05
  posted: 2022-03-30
categories:
  - bare-metal
---

The configuration of a flexible IPv6 depends on the Linux distribution running on your Elastic Metal server. This guide covers the steps for configuring your flexible IPv6 on machines running Debian, Ubuntu, or CentOS.

<Macro id="requirements" />
- A Scaleway account logged into the [console](https://console.scaleway.com)

> - [Owner](/identity-and-access-management/iam/concepts/#owner) status or appropriate [IAM permissions](/identity-and-access-management/iam/concepts/#permission) for actions in the intended Organization

Reviewer: ```suggestion
- [Owner](/identity-and-access-management/iam/concepts/#owner) status or [IAM permissions](/identity-and-access-management/iam/concepts/#permission) allowing you to perform actions in the intended Organization
```

  On Debian 9, the network interface is initialized with the command `allow-hotplug` by default. If the network restart fails with this configuration, initialize the network with `auto` to avoid the problem.
</Message>

## How to configure IPv6 on Ubuntu

Ubuntu uses [netplan](https://netplan.io) for network configuration since Ubuntu 18.04 LTS. Netplan configuration files are written in [YAML](https://yaml.org) and located in the `/etc/netplan` directory. Refer to the official [netplan documentation](https://netplan.readthedocs.io/en/stable/examples/) for more information.

1. Open the default configuration file `/etc/netplan/01-netcfg.yaml` in a text editor and edit it as follows:
    ```yaml
    # This is the network config written by 'subiquity'
    network:
      ethernets:
        eno1:
          critical: true
          dhcp-identifier: mac
          dhcp4: true
          addresses: [ "IPV6_ADDRESS/64" ]
          nameservers:
            addresses:
              - "2001:bc8:1008:1::16"
              - "2001:bc8:1008:1::16"
            search:
              - online.net
      version: 2
    ```
    - Replace `eno1` with the correct name of your internet interface.

>     - Replace the [IP addresses of the resolving DNS servers (nameservers)](/console/account/reference-content/scaleway-network-information/#dns-cache-servers) with the ones located in the same geographical location as your machine for b est perfomances.

Reviewer: ```suggestion
    - Replace the [IP addresses of the resolving DNS servers (nameservers)](/console/account/reference-content/scaleway-network-information/#dns-cache-servers) with the ones located in the same geographical location as your machine for best perfomances.
```
